code/modules/surgery/bodyparts/bodyparts.dm

```diff
@@ -124,6 +127,38 @@
 	for(var/obj/item/I in src)
 		I.forceMove(T)
 
+/obj/item/bodypart/proc/can_break_bone()
+	if(broken)
+		return 0
+	if(status == BODYPART_ROBOTIC)
+		return 0
+	if(!has_bones)
+		return 0
+	return 1
+
+/obj/item/bodypart/proc/break_bone()
+	if(!can_break_bone())
+		return
+	broken = TRUE
+	spawn(1) // Delay so it happens after the punch message
+		owner.visible_message("<span class='userdanger'>You hear a cracking sound coming from [owner]'s [parse_zone(src)].</span>", "<span class='warning'>You feel something crack in your [parse_zone(src)]!</span>", "<span class='warning'>You hear an awful cracking sound.</span>")
```

Comment:
I cant think of any way to put this without making attack code super expensive, which is in no way ideal

Reply:
https://github.com/tgstation/tgstation/blob/f01f3767308683afcdfdf052d97773bdf716db99/code/_onclick/item_attack.dm#L102-L105

https://github.com/tgstation/tgstation/blob/e13330fa9565b5b391ddd451cb0782b472c65a21/code/datums/martial.dm#L67-L70

it looks like it would do this already? both have the punch/attackverb PRIOR to apply_damage (which bleeds into receive_damage which bleeds into broken bones)?